### ISSUE TITLE
bug: Permission denied error when trying to specify junit-xml

### ISSUE THREAD

**Issue by matteus6007** [schemathesis/schemathesis#1995]
When I add the command `--junit-xml /testresults/junit-contracts.xml` to generate a junit report I get the following error:

```shell
Error: Could not open file '/testresults/junit-contracts.xml': No such file or directory
```

Full stack trace:

```shell
Traceback (most recent call last):
  File "/home/runner/work/MyDomain.Api.Template/MyDomain.Api.Template/env/lib/python3.10/site-packages/schemathesis/cli/__init__.py", line 1402, in execute
    handler.handle_event(execution_context, event)
  File "/home/runner/work/MyDomain.Api.Template/MyDomain.Api.Template/env/lib/python3.10/site-packages/schemathesis/cli/junitxml.py", line 43, in handle_event
    to_xml_report_file(file_descriptor=self.file_handle, test_suites=test_suites, prettyprint=True)
  File "/home/runner/work/MyDomain.Api.Template/MyDomain.Api.Template/env/lib/python3.10/site-packages/junit_xml/__init__.py", line 322, in to_xml_report_file
    file_descriptor.write(xml_string)
  File "/home/runner/work/MyDomain.Api.Template/MyDomain.Api.Template/env/lib/python3.10/site-packages/click/utils.py", line 141, in __getattr__
    return getattr(self.open(), name)
  File "/home/runner/work/MyDomain.Api.Template/MyDomain.Api.Template/env/lib/python3.10/site-packages/click/utils.py", line 162, in open
    raise FileError(self.name, hint=e.strerror) from e
click.exceptions.FileError: No such file or directory
```

**Comment by Stranger6667** [comment]
I moved it :) will take a deeper look at it over the weekend

**Comment by Stranger6667** [comment]
Will be fixed by #1999. Additionally, I'll update the handling of permission errors, so they appear more clear in the output
